parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileWriter.java

```diff
@@ -493,6 +492,9 @@ public void writeDictionaryPage(DictionaryPage dictionaryPage,
     dictionaryPage.getBytes().writeAllTo(out); // for encrypted column, dictionary page bytes are already encrypted
     encodingStatsBuilder.addDictEncoding(dictionaryPage.getEncoding());
     currentEncodings.add(dictionaryPage.getEncoding());
+
+    // First data page comes just after the dictionary page
+    currentChunkFirstDataPage = out.getPos();
```

Comment:
The assumption is data page is immediately written after the dictionary page always. This could be true today but later on, if that assumption is changed, it would be hard to find this line of code that need to be changed.

Reply:
I'll move setting this value to data page writes.

Reply:
Sounds good!